### PR TITLE
Bluetooth: bsim: deduplicate compile scripts

### PR DESCRIPTION
**Pull request by alwa-nordic**
Some tests have a compile script for that test alone, named `_compile.sh` or `_build.sh`. These scripts are useful for developers to avoid compiling the world. But these scripts were not run in the CI.

These scripts were previously not suitable for running in the CI because would put the artefact in the source directory, but this is no longer the case, and the CI could run these scripts now.

This commit renames any `<test>/test_scripts/_compile.sh` and `<test>/test_scripts/_build.sh` to `<test>/compile.sh`. The test runner in CI ignores any file named `compile.sh`, so there is no need for a underscore prefix.

This fits into and extends the hierarchical structure of the `compile.sh` CI build script tree that starts at `:/test/bsim/compile.sh`.

The newly moved scripts will now be a part of the build script tree, called from a `compile.sh` where a duplicate set of build commands was before. The sub scripts use `run_in_background` to preserve the parallelism as it was before this change.

This removes an error-prone duplication and makes the CI test all the build scripts.

Tests that did not have any compile script are not given one for now.

### CODE REVIEW

**Review by aescolar — COMMENTED** [review]
Fine by me

**Review by jhedberg — APPROVED** [review]

File: tests/bsim/bluetooth/host/compile.sh [hunk 1]
```diff
@@ -23,23 +23,19 @@ app=tests/bsim/bluetooth/host/iso/bis compile
...
-app=tests/bsim/bluetooth/host/privacy/central compile
-app=tests/bsim/bluetooth/host/privacy/peripheral compile
-app=tests/bsim/bluetooth/host/privacy/peripheral conf_file=prj_rpa_expired.conf compile
-app=tests/bsim/bluetooth/host/privacy/peripheral conf_file=prj_rpa_sharing.conf compile
-app=tests/bsim/bluetooth/host/privacy/device compile
-app=tests/bsim/bluetooth/host/privacy/legacy compile
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/privacy/central/compile.sh
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/privacy/peripheral/compile.sh
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/privacy/device/compile.sh
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/privacy/legacy/compile.sh
```
Comment (Thalley) [inline tests/bsim/bluetooth/host/compile.sh:37]: Is a followup PR planned to increase consistency, so we don't have some doing `app=` and some doing `run_in_background`?

File: tests/bsim/bluetooth/host/security/compile.sh [hunk 2]
```diff
@@ -10,13 +10,13 @@ set -ue
...
-app=tests/bsim/bluetooth/host/security/id_addr_update/peripheral compile
-app=tests/bsim/bluetooth/host/security/security_changed_callback compile
+run_in_background \
+    ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/security/bond_overwrite_allowed/compile.sh
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/security/bond_overwrite_denied/compile.sh
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/security/bond_per_connection/compile.sh
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/security/ccc_update/compile.sh
+run_in_background ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/security/id_addr_update/compile.sh
+run_in_background \
+    ${ZEPHYR_BASE}/tests/bsim/bluetooth/host/security/security_changed_callback/compile.sh
```
Comment (Thalley) [inline tests/bsim/bluetooth/host/security/compile.sh:20]: Would relative paths work in these cases? e.g. something like `run_in_background ./bond_overwrite_allowed/compile.s`

**Review by Thalley — APPROVED** [review]

**Review by jori-nordic — APPROVED** [review]

File: tests/bsim/bluetooth/host/compile.sh (hunk 1 above)
Comment (alwa-nordic) [reply tests/bsim/bluetooth/host/compile.sh:37]: I would like to, yes. I'll open a new issue so it can be discussed there.

Comment (alwa-nordic) [reply tests/bsim/bluetooth/host/compile.sh:37]: https://github.com/zephyrproject-rtos/zephyr/issues/77055

File: tests/bsim/bluetooth/host/security/compile.sh (hunk 2 above)
Comment (alwa-nordic) [reply tests/bsim/bluetooth/host/security/compile.sh:20]: Yes, its possible. But I was asked by @aescolar to not do this in this PR, for consistency, e.g. with `tests/bluetooth/compile.sh`. I'm not planning a followup unless @aescolar agrees.

**Review by aescolar — APPROVED** [review]

**Pull request merged**